tests/testthat/test-api.R

```diff
@@ -0,0 +1,190 @@
+context("api")
+
+test_that("endpoint_baseline_individual", {
+  endpoint <- endpoint_baseline_individual()
+  response <- endpoint$run(readLines("payload/validate_pjnz_payload.json"))
+
+  expect_equal(response$status_code, 200)
+  expect_null(response$error)
+  expect_equal(response$data$hash, scalar("12345"))
+  expect_equal(response$data$data$country, scalar("Malawi"))
+  expect_equal(response$data$data$iso3, scalar("MWI"))
+  expect_equal(response$data$filename, scalar("Malawi2019.PJNZ"))
+  expect_equal(response$data$filters, scalar_null())
+})
+
+test_that("endpoint_baseline_individual works", {
+  api <- api_build()
+  res <- api$request("POST", "/validate/baseline-individual",
+                     body = readLines("payload/validate_pjnz_payload.json"))
+  expect_equal(res$status, 200)
+  body <- jsonlite::fromJSON(res$body)
+  expect_equal(body$data$hash, "12345")
+  expect_equal(body$data$data$country, "Malawi")
+  expect_equal(body$data$data$iso3, "MWI")
+  expect_equal(body$data$filename, "Malawi2019.PJNZ")
+  expect_equal(body$data$filters, NULL)
+})
+
+test_that("endpoint_model_submit can be run", {
+  test_redis_available()
+  queue <- hintr:::Queue$new()
+  endpoint <- endpoint_model_submit(queue)
+  path <- setup_submit_payload()
+  response <- endpoint$run(readLines(path))
+
+  expect_equal(response$status_code, 200)
+  expect_null(response$error)
+  expect_true(!is.null(response$data$id))
+})
+
+test_that("api can call endpoint_model_submit", {
+  test_redis_available()
+  queue <- hintr:::Queue$new()
+  api <- api_build(queue)
+  path <- setup_submit_payload()
+  res <- api$request("POST", "/model/submit",
+                     body = readLines(path))
+  expect_equal(res$status, 200)
+  body <- jsonlite::fromJSON(res$body)
+  expect_equal(body$status, "success")
+  expect_null(body$errors)
+  expect_true(!is.null(body$data$id))
+})
+
+test_that("endpoint_model_status can be run", {
+  test_redis_available()
+  queue <- hintr:::Queue$new()
+  model_run <- endpoint_model_submit(queue)
+  path <- setup_submit_payload()
+  run_response <- model_run$run(readLines(path))
+  expect_equal(run_response$status_code, 200)
+  expect_true(!is.null(run_response$data$id))
+
+  endpoint <- endpoint_model_status(queue)
+  id <- run_response$data$id
+  class(id) <- NULL
```

Comment:
I'm doing this in a couple of places which isn't amazing. This is because if you pass a classed ID into e.g. endpoint$run() then you will get an id in the response with classes `"scalar", "scalar", "character"` so equality checks with input ID and returned ID fail because one has `scalar` class twice. 

It will only be an issue in testing but perhaps worth having pkgapi only add `scalar` if it is not already a `scalar`?

Reply:
Yeah, I think tweaking `scalar` is a good shout